### PR TITLE
Add DotNetListPackage alias for dotnet list package command

### LINKED ISSUE #4224: Add DotNetListPackage alias for dotnet list package command
`dotnet list package` - Lists the package references for a project or solution.

The `dotnet list package` command provides a convenient option to list all NuGet package references for a specific project or a solution.

Docs: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-list-package

### PR DESCRIPTION
**Pull request by Marusyk**
Add an alias for `dotnet list package` command

Fixes #4224

Here is the basic implementation. I need your suggestion on what we should do with the output.
If the format is `console` we will receive smth like this
image
- hard to parse into model.
If the format is json, there could be a several different output which is dependent on the input params
image
image
when error
image

If we allow only JSON, then I'll be able to create models and deserialize (for `console` format it is almost impossible) and return the result for `public ListResult DotNetListPackage(...)` but what if someone just wants to get the output and that's all. 
What do you think?

Thank you in advance

### CODE REVIEW

**Comment by devlead** [comment]
@Marusyk going the JSON route seems like the best option and an object model provides the most value in a Cake context.

**Comment by devlead** [comment]
> ok, so should we allow only `--format json` and have the alias like `public ListPackage List(Project, Settings)`?

Yes, think that makes sense.

**Comment by Marusyk** [comment]
Can someone help me find the model for that json result? 
Depending on the parameters, the result is different every time. I would like to know all the possible attributes that can be returned and make a model for them.

**Comment by devlead** [comment]
Thin it's just a wrapper around NuGet.CommandLine.XPlat which probably means the code is somewhere around here

https://github.com/NuGet/NuGet.Client/tree/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/ListPackage

**Comment by Marusyk** [comment]
Useful: https://github.com/NuGet/Home/blob/dev/accepted/2022/DotnetListPackageMachineReadableJsonOutput.md
It's not quite written in a schema way, but it contains all the examples.

**Comment by Marusyk** [comment]
thanks @augustoproiete . can you help me? I can't figure out why the tests are failing

**Comment by Marusyk** [comment]
can anyone help me to fix this PR?)

**Comment by devlead** [comment]
> can anyone help me to fix this PR?)

Rebased against latest, fixed the failing unit tests and integration test (needed to ensure project restored before executing as integration test cleans between tfms).

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@Marusyk your changes have been merged, thanks for your contribution 👍
